### PR TITLE
Task: Fullstack developer

### PR DESCRIPTION
**Pull request by arparthasarathi**
This PR contains the backend codebase of the task provided for the purpose of evaluation.

The details of the task provided can be found [here](https://docs.google.com/document/u/2/d/e/2PACX-1vSQMOtWnw4BuAQ1mHq1PmeICRtIf-FdsgjIJX1dNJRiZnT9yQYWuWNfbHAKk7pygGL2DJVVcG9lKgj5/pub)

This is a pure Rails 5 API deployed at the following URL

https://seek-expert-advice.herokuapp.com

Some of the responses of the API can be tested here

**List all posts**: https://seek-expert-advice.herokuapp.com/api/v1/posts

**Show a post:** https://seek-expert-advice.herokuapp.com/api/v1/posts/corrupti-voluptatibus-nobis-recusandae

Note: The database has been populated with test data of 5 users, 50 posts and 5 answers for each posts along with the tags.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  #
+  #  @return [nil]
+  def assign_account(account_id)
+    self.account_id = account_id
+  end
+
+  ## Increment the views count by 1 for a post instance
+  #
+  #  @return [nil]
+  def increment_views_count
```
Comment (matixmatix) [inline app/models/post.rb]: Rails has a handy method for that: https://stackoverflow.com/questions/11499110/increment-vs-1

File: app/serializers/user_serializer.rb [hunk 2]
```diff
@@ -3,6 +3,6 @@ class UserSerializer < ActiveModel::Serializer
 
   def account
     account_user = @instance_options[:account_user]
-    account_user.account
+    account_user ? account_user.account : nil
```
Comment (matixmatix) [inline app/serializers/user_serializer.rb]: `account_user&.account`

File: db/migrate/20190519170921_create_posts.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class CreatePosts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :posts do |t|
```
Comment (matixmatix) [inline db/migrate/20190519170921_create_posts.rb]: Missing indexes

File: db/schema.rb [hunk 4]
```diff
@@ -61,6 +61,49 @@
...
+    t.string "title"
+    t.string "slug"
+    t.text "body"
+    t.bigint "user_id", null: false
+    t.bigint "account_id", null: false
+    t.integer "question_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "views_count", default: 0
+<<<<<<< HEAD
```
Comment (matixmatix) [inline db/schema.rb]: 👎

File: app/models/post.rb (hunk 1 above)
Comment (arparthasarathi) [reply app/models/post.rb]: I wasn't aware of the handy method actually. But even when using the method mentioned above, you need to call the `save` method to update the attribute. Changed the code that manually increments the value to this handy method in my code

File: app/serializers/user_serializer.rb (hunk 2 above)
Comment (arparthasarathi) [reply app/serializers/user_serializer.rb]: Safe navigation operator 👍

File: db/migrate/20190519170921_create_posts.rb (hunk 3 above)
Comment (arparthasarathi) [reply db/migrate/20190519170921_create_posts.rb]: Added indices and updated the database. Thanks for pointing out!

File: db/schema.rb (hunk 4 above)
Comment (arparthasarathi) [reply db/schema.rb]: Shit. Missed this due to the last minute urgency to complete the task. I shall make sure that this doesn't happen again.

**Pull request closed without merging**